students/TinaB/lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,231 @@
+                    break
+            else: # firstname is not in list
+                print("HERE!")
+                first_name = fname
+                last_name = input(
+                    "No donor with that first name found in donor list. Please provide a last name to add donor to list: ")
+                donor_return.append(first_name)
+                donor_return.append(last_name)
+                break
+
+        else:  # user input = 2
+            if lname.lower() in donor["lastname"].lower():
+                donor_check = input(
+                    "Is this the donor you are looking for:  {} {}?   Please enter yes or no:  ".format(donor["firstname"], donor["lastname"]))
+                if donor_check == "yes":
+                    donor_return.append(donor["firstname"])
+                    donor_return.append(donor["lastname"])
+                    donor_check = 1
+                    break
+                else:
+                    last_name = lname
+                    first_name = input(
+                        "No donor with that last name found in donor list. Please provide a first name for addition to donor list")
+                    donor_return.append(first_name)
+                    donor_return.append(last_name)
+                    break
+
+    return donor_return, donor_check
+
+
+def send_single_thank_you():
+    """ function for sending thank you message"""
+    donor_name, donor_check = get_name_input()
+    donor_amount = float(input('Please enter a donation amount for {} {}: '.format(
+        donor_name[0], donor_name[1])))
+
+    # search through donors with returned name - 1 means name is in list, 0 means not in list
+    if donor_check == 1:
+        for donor in donors_list_dictionary:
+            if donor["firstname"] == donor_name[0] and donor["lastname"] == donor_name[1]:
+                donor["donations"].append(donor_amount)
+                donor["donate_times"]+=1
+                thank_you_letter = print_thank_you(
+                    donor["firstname"], donor_amount)
+                print(thank_you_letter)
+                break
+    else:
+        # add new donor to list
+        donors_list_dictionary.append(
+            {"firstname": donor_name[0], "lastname": donor_name[1], "donations": [donor_amount], "donate_times": 1})
+        thank_you_letter = print_thank_you(donor_name[0], donor_amount)
+        print(thank_you_letter)
+
+
+def print_donors_names():
+    """ prints list of donors"""
+    print()
+    for d in donors_list_dictionary:
+        print("{firstname} {lastname}".format(**d))
+    print()
+
+
+def print_thank_you(donor_fname, amount):
+    """ prints thank you message"""
+    d = {'name': donor_fname, 'donation': amount}
+    thank_you = '\nDear {},\n '.format(donor_fname)
+    thank_you += '\tThank you for your generous donation of ${donation:,.2f}\n'.format(
+        **d)
+    thank_you += 'Sincerely, \nThe ChickTech Donations Department\n'
```

Comment:
It's generally preferred to use string formatting on the entire string (like you're doing in the `print_thank_you_total` function below) rather than concatenate